multiraft/multiraft.go

```diff
@@ -654,10 +654,21 @@ func (s *state) createGroup(groupID uint64) error {
 
 func (s *state) removeGroup(op *removeGroupOp) {
 	// Group creation is lazy and idempotent; so is removal.
-	if _, ok := s.groups[op.groupID]; !ok {
+	log.V(3).Infof("removing group %d", op.groupID)
+	g, ok := s.groups[op.groupID]
+	if !ok {
 		op.ch <- nil
 		return
 	}
+
+	// Cancel commands which are still in transit.
+	for cmdID, prop := range g.pending {
+		if prop.ch != nil {
+			prop.ch <- util.Errorf("group deleted")
```

Comment:
ok, I'll do it tomorrow. BTW looks that now that this bug is fixed, we'll hit #906 more often (at least it happened twice today).